### PR TITLE
test: add test case validate completed_item when new leader add

### PR DESCRIPTION
**Pull request by glepnir**
Problem: missing test case the completed_item validate when new leader.

Solution: add a test case.

### CODE REVIEW

File: test/functional/editor/completion_spec.lua [hunk 1]
```diff
@@ -1050,6 +1050,47 @@ describe('completion', function()
...
+        endfunc
+        set omnifunc=Omni_test
+        set complete=.
+        function! OnChange()
+          let g:event = copy(v:event)
+          let g:item = get(v:event, 'completed_item', {})
+          let g:word = get(g:item, 'word', v:null)
+        endfunction
+        autocmd CompleteChanged * call OnChange()
+      ]])
```
Comment (zeertzjq) [inline test/functional/editor/completion_spec.lua:1069]:
```suggestion
      funct Omni_test(findstart, base)
        if a:findstart
          return col(".")
        endif
        return [#{word: "one"}, #{word: "two"}, #{word: "five"}]
      endfunc
      set omnifunc=Omni_test
      set complete=.
      function! OnChange()
        let g:event = copy(v:event)
        let g:item = get(v:event, 'completed_item', {})
        let g:word = get(g:item, 'word', v:null)
      endfunction
      autocmd CompleteChanged * call OnChange()
    ]])
```

Comment (glepnir) [reply test/functional/editor/completion_spec.lua]: seems like can't use stylua on test file T.T

File: test/functional/editor/completion_spec.lua [hunk 2]
```diff
@@ -1050,6 +1050,47 @@ describe('completion', function()
     feed('<esc>')
   end)
 
+  it('set completed_item correct when new leader add', function()
```
Comment (zeertzjq) [inline test/functional/editor/completion_spec.lua:1053]:
```suggestion
  it('sets completed_item correctly when adding new leader', function()
```

**Comment by zeertzjq** [comment]
Isn't this the same as the test in #27332?

**Comment by glepnir** [comment]
same. but lua with screen expect. don't needed ?

**Pull request closed without merging**